src/metatrain/experimental/soap_bpnn/trainer.py

```diff
@@ -39,6 +39,9 @@ def __init__(self, train_hypers):
         self.optimizer_state_dict = None
         self.scheduler_state_dict = None
         self.epoch = None
+        self.best_loss = None
+        self.best_model_state_dict = None
+        self.best_optimizer_state_dict = None
```

Comment:
These look like changes to soap-bpnn, should these be in a separate PR?

Reply:
They're related, this is slightly adapting the training procedure for SOAP-BPNN to follow that of PET, which is a bit better